src/plugins/header/options/options.ts

```diff
@@ -54,19 +54,19 @@ export class UIPOptions extends UIPPlugin {
 
     const config = UIPOptions.UIPOptionsConfig.find((elem) => elem.attrName === attrName);
     if (!config) return;
-    if (newVal === null) {
+    if (newVal === null) {  // add option icon
       this.render([config]);
-    } else {
-      const option = this.options.get(config.rootControlledAttr);
+    } else { // remove option icon
+      const option = this.options.get(config.optionValue);
       if (!option) return;
-      this.options.delete(config.rootControlledAttr);
+      this.options.delete(config.optionValue);
       option.remove();
     }
```

Comment:
Let's do it like this 
```
const config: OptionConfig | undefined = UIPOptions.UIPOptionsConfig.find(elem => elem.attrName === attrName);
    if (!config) return;

    if (newVal === null) this.render([config]); // add option icon
    if (newVal !== null) { // remove option icon
      const option = this.options.get(config.optionValue);
      if (!option) return;
      this.options.delete(config.optionValue);
      option.remove();
    }
```
And btw the name `optionValue` for OptionConfig is very nice to me!